### PR TITLE
Fix retry mechanism embedding component 

### PR DESCRIPTION
**Pull request by PhilippeMoussalli**
Reduce from infinite retires to a fix number

### CODE REVIEW

File: components/embed_text/src/main.py [hunk 1]
```diff
@@ -61,9 +63,13 @@ def get_embedding_model(
...
 
-    @retry()  # make sure to keep trying even when api call limit is reached
+    # make sure to keep trying even when api call limit is reached
     def get_embeddings_vectors(self, texts):
-        return self.embedding_model.embed_documents(texts.tolist())
+        @retry(tries=self.tries, logger=logger)
+        def return_embeddings():
+            return self.embedding_model.embed_documents(texts.tolist())
+
+        return return_embeddings()
```
Comment (RobbeSneyders) [inline components/embed_text/src/main.py:72]: FYI, you can just use a decorator like a normal function.

```suggestion
        retry_wrapper = retry(tries=self.tries, logger=logger)
        return retry_wrapper(self.embedding_model.embed_documents(texts.tolist()))
```

**Review by RobbeSneyders — COMMENTED** [review]
Thanks @PhilippeMoussalli, we should indeed not do this endlessly.

File: components/embed_text/src/main.py (hunk 1 above)
Comment (PhilippeMoussalli) [reply components/embed_text/src/main.py]: 🤯

File: components/embed_text/fondant_component.yaml [hunk 2]
```diff
@@ -40,6 +40,10 @@ args:
       Additional keyword arguments required for api initialization/authentication.
     type: dict
     default: {}
+  tries:
```
[the remark this answers is not in the document]
Comment (PhilippeMoussalli) [reply components/embed_text/fondant_component.yaml]: sure changed it :)

**Review by RobbeSneyders — APPROVED** [review]

**Pull request merged**